### PR TITLE
[WIP]: SVD

### PR DESCRIPTION
**Pull request by HannesHolste**
SVD, delegating to scipy svd (LAPACK).

Open questions that need to be addressed:

@antgonza / @josenavas: 
1) Is it correct to only return the real number part of the eigenvectors/eigenvalues? I'm not sure how the output will be used ie if you want to keep the imaginary part.

`eigvecs = np.array(U.real)`

2) For the unit test, I want to ensure that the returned `coords, eigvals, percentages` are correct. Jose suggested looking into unit tests in scipy/sklearn. I did that but I'm cautious about simply using sklearn.pca on the same distance matrix and then comparing my output with theirs using `assertAlmostEqual`, because they do some funky stuff to the eigenvectors:
https://github.com/scikit-learn/scikit-learn/blob/51a765a/sklearn/decomposition/pca.py#L249

Any advice on how to properly unit test this, and the other methods I'm implementing? After a lot of reading and browsing I'm stuck on figuring out a good way to sanity-check my work.

3) For `pcnts` (percentage variation array), we can either represent them from 0.0 to 1.0 or 0-100. Is there a standard for this? The other methods like nystrom didn't calculate percentages at all so I couldn't follow the standard. 

4) In general, do you want methods to return standard python arrays, or to return and pass around numpy arrays? (currently returning numpy arrays)

### CODE REVIEW

**Comment by antgonza** [comment]
Just real is fine. I'm not sure we should test the output as that's responsibility of the library, however we should check that we format that output correctly for our tools ...

**Comment by antgonza** [comment]
Expanding on my last reply:
1. Yes, just real. I remember having this discussion over the actual implementation we currently have, you can follow here: https://github.com/biocore/scikit-bio/pull/90/files#diff-d5b2dfbdce30869718db88105cd9fdd2R18
2. The biggest issue with unit testing is that depending on the hardware and software stack you are using the values/signs of the results might change. For example, from the same distance matrix we could get 2 flipped versions of the same plot in 2 different machines; this is a common thing during workshops. With this in mind, I would not test actual values from those methods coming from other libraries, just that we are parsing them correctly, and I would put really simple distance matrices for those other methods we are denovo implementing here.
3. We like 0-100 and for those methods without values we simply add N/A. Some of these methods "improve" their performance by not calculating the pcnts or simply is not possible due to their algorithms.
4. I think we have always return numpy arrays. I think this is fine.

**Comment by ElDeveloper** [comment]
> 1. The biggest issue with unit testing is that depending on the hardware and software stack you are using the values/signs of the results might change. For example, from the same distance matrix we could get 2 flipped versions of the same plot in 2 different machines; this is a common thing during workshops. With this in mind, I would not test actual values from those methods coming from other libraries, just that we are parsing them correctly, and I would put really simple distance matrices for those other methods we are denovo implementing here.

I know this is just an example, but this was very cleanly solved by
@Jorge-C in skbio:
https://github.com/biocore/scikit-bio/blob/master/skbio/util/_testing.py#L248

**Comment by antgonza** [comment]
Good point @ElDeveloper, we should also add skbio to avoid duplication of code.

File: mdsa/approximate_mds.py [hunk 1]
```diff
@@ -161,6 +164,23 @@ def principal_coordinates_analysis(distance_matrix, algorithm, dimensions):
...
+        eigvecs = np.array(U.real)
+
+        # for our output coordinates, only pick the desired dimensions
+        coords = eigvecs[:, :dimensions]
+
+        # eigenvalues without the imaginary component
+        eigvals = np.array(s.real)[:dimensions]
+
+        # percentages (how much variation each coord is explaining)
+        pcnts = ((eigvals / np.sum(eigvals)) * 100)[:dimensions]
```
Comment (HannesHolste) [inline mdsa/approximate_mds.py]: note to self: return full solution (do not reduce to k dimensions). 

Also raise a warning that says: dimensions parameter ignored for this method (python warnings module: warnings.warn)

**Comment by antgonza** [comment]
One minor flake8 error.

**Pull request closed without merging**